### PR TITLE
[RDY] Update PR template and Contributing

### PR DESCRIPTION
**Pull request by lewri**
Github Projects can be utilised by contributors now to point their own PRs into specific projects. This helps with looking over at the end of a round to make release notes.
There are also new PR Labels to help classify PRs too.

### CODE REVIEW

**Comment by TheCycoONE** [comment]
Can you point people to the draft feature as well.

File: CONTRIBUTING.md
Comment (Alberth289346) [inline CONTRIBUTING.md:39]: Add a sentence that pull requests without label are considered ready for review and merge if found ok.

File: CONTRIBUTING.md
Comment (Alberth289346) [inline CONTRIBUTING.md:17]: Suggest to ask or discuss in the Matrix channel as well in case of not being sure the new feature is wanted?

File: CONTRIBUTING.md
Comment (Alberth289346) [inline CONTRIBUTING.md:19]: Insert empty line

File: CONTRIBUTING.md
Comment (Alberth289346) [inline CONTRIBUTING.md:42]: Here too an empty line

Comment (lewri) [reply CONTRIBUTING.md:42]: Backspace ``\`` is an escape character in MD, so newline not needed.

File: CONTRIBUTING.md
Comment (Alberth289346) [inline CONTRIBUTING.md:47]: 4a Fork the repository

File: CONTRIBUTING.md
Comment (Alberth289346) [inline CONTRIBUTING.md:54]: first  `git checkout master` (although you need that only after fetching)

File: CONTRIBUTING.md
Comment (Alberth289346) [inline CONTRIBUTING.md:55]: `git merge --ff-only upstream/master` rather than `rebase` as the latter may introduce addition commits that don't exist in upstream, thus making a mess when you push a branch

File: CONTRIBUTING.md
Comment (Alberth289346) [inline CONTRIBUTING.md:62]: You branch from your own master normally

``` shell
git checkout master
git branch myfeature
```
(and replace `myfeature` by a more descriptive line like `add-rats-levels`)

File: CONTRIBUTING.md
Comment (Alberth289346) [inline CONTRIBUTING.md:80]:
```suggestion
If your feature is very big, break it into more smaller steps and do a separate commit
```

File: CONTRIBUTING.md
Comment (Alberth289346) [inline CONTRIBUTING.md:86]:
```suggestion
there might be other important changes that other people are make to CorsixTH.  It
```
(?)

File: CONTRIBUTING.md
Comment (TheCycoONE) [reply CONTRIBUTING.md:39]: I don't need the [RDY] label anymore, we can drop that as a concept - it just clutters the git history with [RDY] 😄

File: CONTRIBUTING.md
Comment (TheCycoONE) [reply CONTRIBUTING.md:47]: Step 2 was frok

File: CONTRIBUTING.md
Comment (Alberth289346) [inline CONTRIBUTING.md:89]: Again first `git checkout master`, since you sync `upstream/master` to your own `master`.
Also not `rebase` but `merge --ff-only` as before.

Then update your branch to the new master
``` shell
git checkout myfeature
git rebase master
```
and resolve any cpnflicts that may arise at this point.

File: CONTRIBUTING.md
Comment (TheCycoONE) [reply CONTRIBUTING.md:80]: without the redundant `more` though.

File: CONTRIBUTING.md
Comment (TheCycoONE) [reply CONTRIBUTING.md:86]: I don't think so Albert, doesn't read like English.

File: CONTRIBUTING.md
Comment (lewri) [reply CONTRIBUTING.md:17]: That's the point of submitting a Feature Request  as below

Comment (lewri) [reply CONTRIBUTING.md:17]: But you're right, just adding an enhancement PR without reason could be peculiar. Not that we could stop people doing it. Perhaps instead we should suggest making a Feature Request first.

File: CONTRIBUTING.md
Comment (lewri) [reply CONTRIBUTING.md:39]: I would disagree -- RDY is very handy as a concept. I know exactly what state a PR is in without going into it.

File: CONTRIBUTING.md
Comment (TheCycoONE) [reply CONTRIBUTING.md:17]: I think we do want to stop people from doing it, drive by feature PRs for things we haven't discussed tend to waste everyone's time.

File: CONTRIBUTING.md
Comment (lewri) [reply CONTRIBUTING.md:19]: Again note the backspace

File: CONTRIBUTING.md
Comment (lewri) [reply CONTRIBUTING.md:39]: (Not that people listen to us 🙃 anyway)

File: CONTRIBUTING.md
Comment (lewri) [reply CONTRIBUTING.md:54]: Can you add your git changes as suggestions for me so I don't mess their order up?

File: CONTRIBUTING.md
Comment (lewri) [reply CONTRIBUTING.md:80]:
```suggestion
If your feature is very big, break it into smaller steps and do a separate commit
```

File: CONTRIBUTING.md
Comment (TheCycoONE) [reply CONTRIBUTING.md:55]: `git config branch.master.mergeOptions --ff-only` once, and never specify it again

File: CONTRIBUTING.md
Comment (TheCycoONE) [reply CONTRIBUTING.md:62]: Either is fine.
`git checkout -b myfeature master`

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:18]:
```suggestion
please open a new issue for it and tell other devs here that your going to/have started working on it.

**If you have an idea for an enhancement that does not have a discussion, please make a *Feature Request* first to discuss its desirability.  This helps make sure your development time is a worthy endeavour.**

## Enhancement/Suggestions ##
```

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:31]:
```suggestion
CorsixTH uses the following labels inside your Pull Request's title to denote its current progress. Unlabelled pull request will be considered *Ready for Review* and may be merged without warning.
```

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:74]:
```suggestion
4. Create and checkout your feature branch
	- ``git checkout -b myfeature master``
5. Make your changes
6. Unittest continuously, see README.txt in CorsixTH/Luatest for more info
7. Review your changes
	- ``git diff``
	- ``git add`` | for each file
	- ``git diff --check`` | Check there's no white spaces.
	- ``git commit`` | Write an informative commit message and save
8. Push your changes to your fork
	- ``git push origin myfeature``
9. Create a Pull Request -- GitHub will pick up your latest branch committed to automatically when accessing CorsixTH/CorsixTH.
```

File: CONTRIBUTING.md
Comment (Alberth289346) [reply CONTRIBUTING.md:19]: I meant an empty line after the header line and before this line. See also line 4.

Not sure where you expect such a comment if not here.

File: CONTRIBUTING.md
Comment (lewri) [reply CONTRIBUTING.md:89]: Can you write this into a suggestion of the correct use/order?

File: CONTRIBUTING.md
Comment (lewri) [reply CONTRIBUTING.md:55]: Please can you reflect this in a suggestion as I don't understand

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:42]:
```suggestion
*NB: If you're using GitHub Desktop ignore the git specific commands and follow guidance from [GitHub Docs](https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/overview/getting-started-with-github-desktop)*\
```

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:19]:
```suggestion

If you want to make a suggestion, please use the *Feature Request* option when making a new issue. It will provide all the information needed to evaluate the suggestion or enhancement.\
```

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:41]:
```suggestion
#### CONTRIBUTING CODE: ####

```

File: CONTRIBUTING.md
Comment (TheCycoONE) [reply CONTRIBUTING.md:41]: I would rather mark all other states draft, wip, and leave ready blank.

Comment (lewri) [reply CONTRIBUTING.md:41]: What about RFC?

Comment (TheCycoONE) [reply CONTRIBUTING.md:41]: Sure, RFC serves a purpose - please look at me, but don't merge.

Comment (lewri) [reply CONTRIBUTING.md:40]: Done. Assuming nobody is gonna be punished for still using [RDY] if they wanted though. That was my only holdup.

File: CONTRIBUTING.md
Comment (TheCycoONE) [inline CONTRIBUTING.md:36]: Note that you do NOT need to RFC a PR before moving it to ready. The purpose of RFC is to get feedback on a general direction, or ask for assistance before finishing the PR. Ideally a comment would be added at the same time with the question / focus you'd like us to review.

Comment (lewri) [inline CONTRIBUTING.md:36]:
```suggestion
| **[RFC]** | (Optional stage) Request for Comments (aka Ready for Review) |
```

File: .github/pull_request_template.md
Comment (TheCycoONE) [inline .github/pull_request_template.md:3]: Can you drop the `If you can't see this stuff` and possibly clarify who a contributor is. I think you mean a member of CorsixTH, which is a pretty small group of people - maybe drop that whole section?

File: CONTRIBUTING.md
Comment (TheCycoONE) [inline CONTRIBUTING.md:5]: drop `please`

File: CONTRIBUTING.md
Comment (TheCycoONE) [inline CONTRIBUTING.md:12]: Not clear what you mean here. What is a `discussion`?

File: CONTRIBUTING.md
Comment (TheCycoONE) [inline CONTRIBUTING.md:14]: `issue` not `issue discussion`

Comment (TheCycoONE) [reply CONTRIBUTING.md:14]: change out `what you want to work on` for `the bug fix or feature you plan to contribute`

File: CONTRIBUTING.md
Comment (TheCycoONE) [inline CONTRIBUTING.md:15]: drop `please`

File: CONTRIBUTING.md
Comment (TheCycoONE) [inline CONTRIBUTING.md:17]: drop `please`

File: CONTRIBUTING.md
Comment (TheCycoONE) [inline CONTRIBUTING.md:37]: `none`, not `blank`

**Review by TheCycoONE — CHANGES_REQUESTED** [review]
Drop all use of `please`, these are guidelines not requests.
Try to stick to the same terms github uses to describe concepts for clarity
Stick to 80 columns where possible.

File: .github/pull_request_template.md
Comment (lewri) [inline .github/pull_request_template.md:3]:
```suggestion
<!-- If your PR is still being drafted you should submit it as a "Draft Pull Request" or add [WIP] to your title. -->
```

File: CONTRIBUTING.md
Comment (TheCycoONE) [inline CONTRIBUTING.md:119]: replace `and pretends that all your changes were made after them` with `and moves your commits on top of them`

**Review by TheCycoONE — CHANGES_REQUESTED** [review]
In general I think more effort should be put into distinguishing normative and non-normative language. Standards which could cause a PR to be rejected or make extra work for maintainers need to be described clearly as rules.

See:
https://github.com/wesnoth/wesnoth/blob/master/CONTRIBUTING.md as a good example of what I mean. Frequent use of `recommend` and `must` would help.

https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md is an example of succinctly providing all the information needed to contribute and nothing more.

File: .github/pull_request_template.md
Comment (TheCycoONE) [inline .github/pull_request_template.md:1]: *must* not *should*

File: CONTRIBUTING.md
Comment (TheCycoONE) [reply CONTRIBUTING.md:35]: I prefer the wording of Pull Requests from CONTRIBUTING.txt

File: CONTRIBUTING.md
Comment (TheCycoONE) [reply CONTRIBUTING.md:119]: wording predates your change - feel free to ignore

**Comment by lewri** [comment]
I imagine we're not as strict as the examples above but I will take note to link to our coding conventions page in the wiki.

File: .github/pull_request_template.md
Comment (lewri) [reply .github/pull_request_template.md:1]:
```suggestion
<!-- If your PR is still being drafted you must either submit it as a "Draft Pull Request" or add [WIP] to your title. -->
```

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:25]:
```suggestion
To suggest an enhancement, open a new [issue](https://github.com/CorsixTH/CorsixTH/issues/new/choose)
and click the *Feature Request* button.\
It will provide all the information needed to evaluate the
```

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:36]:
```suggestion
link those issues to the description by referencing them with a `#` followed by the issue number. 
The pull request provides a default layout to help you fill out this information.
```

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:49]:
```suggestion
You can also submit a **Draft Pull Request** as an alternative to the
```

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:102]:
```suggestion
If your pull request can not be summarised in a single sentence, break it into smaller steps
and do a separate commit for every independent operation (e.g. one sentence per commit is a
good measure). This means doing step 10 several times as you go. You still only need to do
one pull request at the end.
```

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:108]:
```suggestion
there might be other important changes other people are making to CorsixTH.  To
ensure your pull request operates correctly within the latest changes:
```

File: CONTRIBUTING.md
Comment (lewri) [reply CONTRIBUTING.md:118]:
```suggestion
This downloads all the changes from CorsixTH/CorsixTH since you
```

File: CONTRIBUTING.md
Comment (lewri) [reply CONTRIBUTING.md:119]:
```suggestion
started, and moves your commits on top of them.  If there are
```

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:56]:
```suggestion
#### CONTRIBUTING CODE: ####
**Important!**
Ensure you have read our [Coding Conventions](https://github.com/CorsixTH/CorsixTH/wiki/Coding-Conventions)
before starting your work. CorsixTH may uses styles different to your previous experience with ``Lua``
and ``C++``. The [Wiki](https://github.com/CorsixTH/CorsixTH/wiki/) also contains other useful information
that could be beneficial before contributing code.

*NB: If you're using GitHub Desktop ignore the git specific commands and follow
guidance from [GitHub Docs](https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/overview/getting-started-with-github-desktop)*\
```

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:54]:
```suggestion
**Important!**\
```

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:26]:
```suggestion
suggestion or enhancement.
```

File: CONTRIBUTING.md
Comment (TheCycoONE) [inline CONTRIBUTING.md:19]: Most of this document uses single space after . style.

**Comment by TheCycoONE** [comment]
Looking much better, thanks for your persistence.

I'm not sure if it's worth calling out, but C++ must be formatted with clang-format before it will be accepted.

File: CONTRIBUTING.md
Comment (lewri) [inline CONTRIBUTING.md:1]:
```suggestion
# CorsixTH Contributing Guidelines #
```
Oh dear... how did I not notice this.

**Pull request merged**
